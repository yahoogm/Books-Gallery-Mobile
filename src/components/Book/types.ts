import {VolumeInfo} from '../../types/types';

type DetailBookType = {
  volumeInfo: VolumeInfo;
};

export type BookProps = {
  uriDetailBook: string;
  authorsName: string[] | string;
  categories: string[] | string;
  detailBook: DetailBookType;
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  refReadBook: React.RefObject<any>;
};

export type BookContainerProps = {
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  refReadBook: React.RefObject<any>;
};
