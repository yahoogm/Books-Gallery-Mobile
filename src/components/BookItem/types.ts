export type BookItemProps = {
  detailBook: string | number | string[];
  label: string;
};
