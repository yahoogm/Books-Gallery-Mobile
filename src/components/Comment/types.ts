import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {BookItem, Comment, RootStackParamList} from '../../types/types';
import {ObjectSchema} from 'yup';

export type FormCommentValues = {
  comment: string;
};

export type AddCommentValues = {
  userName: string;
  bookId: string;
  profilePic: string;
  id: string;
  ulasan: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
};

// type for handling navigation to login page
export type NavigationProp = NativeStackNavigationProp<
  RootStackParamList,
  'Login'
>;

export type ToastRenderProps = {
  id: string;
};

export type UserItem = {
  email: string;
  familyName: string;
  givenName: string;
  id: string;
  name: string;
  photo: string;
};

export type HandleBookCommentFunction = {
  handleNavigateUserNotLogin: () => void;
  handleAddCommentBook: (review: AddCommentValues) => void;
  handleDeleteCommentBook: () => void;
  handleUpdateCommentBook: (ulasan: string, updateAt: string) => void;
  getIdDocumentFromFirestore: (id: string) => void;
};

export type CommentProps = {
  refOpenDeleteModal: React.RefObject<any>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  handleButton: HandleBookCommentFunction;
  isEdit: boolean;
  setIsEdit: React.Dispatch<React.SetStateAction<boolean>>;
  commentSchema: ObjectSchema<FormCommentValues>;
  initialValues: FormCommentValues;
  isLogin: boolean;
  user: UserItem;
  detailBook: BookItem;
  commentBook: Array<Comment>;
};
