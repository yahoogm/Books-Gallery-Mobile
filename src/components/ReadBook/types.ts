export type ReadBookProps = {
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  showBook: boolean;
  refReadBook: React.RefObject<any>;
  isbn: number;
};
