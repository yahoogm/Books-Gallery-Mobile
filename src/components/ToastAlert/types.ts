export type ToastAlertProps = {
  id: string;
  message: string;
  action: 'success' | 'info' | 'error' | 'warning' | 'attention';
};
