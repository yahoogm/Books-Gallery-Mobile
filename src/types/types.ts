type VolumeInfo = {
  title: string;
  publishedDate: string;
  description: string;
  authors: Array<string>;
  publisher: string;
  pageCount: number;
  imageLinks: {smallThumbnail: string; thumbnail: string; large?: string};
  categories: Array<string>;
  industryIdentifiers: Array<{identifier: string; type: string}>;
};

export type Review = {
  bookId: string;
  createdAt: {};
  id: string;
  profilePic: string;
  ulasan: string;
  updatedAt: {};
  userId: string;
  userName: string;
};

export type BookItem = {
  id: string;
  volumeInfo: VolumeInfo;
};

export type BookState = {
  type: string;
  reviewId: string;
  detailBook: BookItem;
  books: {
    items: [];
  };
  reviewBook: Array<Review>;
};

export type userItem = {
  email: string;
  familyName: string;
  givenName: string;
  id: string;
  name: string;
  photo: string;
};

export type RootStackParamList = {
  DetailBook: {bookId: string};
  Login: undefined;
};
