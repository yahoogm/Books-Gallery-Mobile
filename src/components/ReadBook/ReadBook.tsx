import {
  ScrollView,
  ModalBackdrop,
  ModalContent,
  ModalBody,
  Modal,
} from '@gluestack-ui/themed';
import WebView from 'react-native-webview';
import {ReadBookProps} from './types';

const ReadBook: React.FC<ReadBookProps> = ({
  refReadBook,
  setShowBook,
  showBook,
  isbn,
}) => {
  return (
    <ScrollView>
      <Modal
        size="lg"
        padding={0}
        isOpen={showBook}
        testID="modalEmbeddedBook"
        onClose={() => {
          setShowBook(false);
        }}
        finalFocusRef={refReadBook}>
        <ModalBackdrop />
        <ModalContent>
          <ModalBody marginTop={8}>
            <WebView
              originWhitelist={['*']}
              style={{width: 'auto', height: 500}}
              source={{
                html: `
                      <meta name="viewport" content="width=device-width">
                      <script type="text/javascript" src="https://www.google.com/books/jsapi.js"></script>
                      <script type="text/javascript">
                        google.books.load();

                        function alertNotFound() {
                          alert("Buku tidak dapat dibaca!");
                        }

                        function initialize() {
                          var viewer = new google.books.DefaultViewer(
                            document.getElementById('viewerCanvas'),
                          );
                          viewer.load('ISBN:${isbn}', alertNotFound);
                        }

                        google.books.setOnLoadCallback(initialize);
                      </script>

                      <body>
                      <div id="viewerCanvas" style="width: 500px; height: 860px"></div>
                    </body>
                    `,
              }}
            />
          </ModalBody>
        </ModalContent>
      </Modal>
    </ScrollView>
  );
};

export default ReadBook;
