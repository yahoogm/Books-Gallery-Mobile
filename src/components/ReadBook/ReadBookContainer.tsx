import ReadBook from './ReadBook';
import {ReadBookProps} from './types';

const ReadBookContainer: React.FC<ReadBookProps> = ({
  refReadBook,
  setShowBook,
  showBook,
  isbn,
}) => {
  return (
    <ReadBook
      refReadBook={refReadBook}
      setShowBook={setShowBook}
      showBook={showBook}
      isbn={isbn}
    />
  );
};

export default ReadBookContainer;
