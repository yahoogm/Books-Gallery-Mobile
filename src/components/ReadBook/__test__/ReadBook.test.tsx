import React, {createRef} from 'react';
import ReadBook from '../ReadBook';
import {ReadBookProps} from '../types';
import {render} from '@testing-library/react-native';

jest.mock('react-native-webview');

describe('Read Book', () => {
  const renderComponent = (props: ReadBookProps) =>
    render(<ReadBook {...props} />);

  it('match snapshot when given props', () => {
    const refReadBook = createRef<any>();
    const mockSetShowBook = jest.fn();
    const props = {
      refReadBook: refReadBook,
      setShowBook: mockSetShowBook,
      showBook: false,
      isbn: 2020400192,
    };

    const tree = renderComponent(props).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not display the modal when showBook is false', () => {
    const refReadBook = createRef<any>();
    const mockSetShowBook = jest.fn();
    const props = {
      refReadBook: refReadBook,
      setShowBook: mockSetShowBook,
      showBook: false,
      isbn: 2020400192,
    };

    const tree = renderComponent(props);
    const instance = tree.queryByTestId('modalEmbeddedBook');

    expect(instance).toBeNull();
  });
});
