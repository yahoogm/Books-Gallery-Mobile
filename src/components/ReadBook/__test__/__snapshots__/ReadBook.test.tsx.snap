// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read Book match snapshot when given props 1`] = `
<RCTScrollView
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      undefined,
    ]
  }
>
  <View />
</RCTScrollView>
`;
